ocaml/xapi/xapi_db_upgrade.ml

```diff
@@ -253,7 +253,10 @@ let upgrade_vm_memory_for_dmc =
           else
             ()
         in
-        List.iter update_vm (Db.VM.get_all_records ~__context)
+        if Pool_features.is_enabled ~__context Features.DMC then
```

Comment:
I don't think this works as expected, as on startup, the DB upgrade runs before the licensing refresh that turns the feature off.

Reply:
Ok - so maybe we need to use the version constraint that is part of the rules mechanism.

Reply:
I'm reducing this to just changing the unit test to use a disabled DMC, as that is the current scenario. The upgrade rule might have to be removed if we backport this code.